src/components/common/NavigationBar.tsx

```diff
@@ -1,42 +1,48 @@
 import styled, { css } from 'styled-components';
 import { COLOR } from '@src/styles/color';
 import { FONT_STYLES } from '@src/styles/fontStyle';
+import { imgLogo } from 'public/assets/images';
 import Link from 'next/link';
 import { useRouter } from 'next/router';
+import { useState } from 'react';
 import ImageDiv from './ImageDiv';
-import { imgLogo } from 'public/assets/images';
+import LoginModal from '../login/LoginModal';
 
 function NavigationBar() {
   const router = useRouter();
+  const [isModalOpen, setIsModalOpen] = useState(false);
 
   return (
-    <StNavigationBar>
-      <Link href="/home">
-        <a>
-          <ImageDiv className="logo" priority src={imgLogo} layout="fill" alt="DeliverBle" />
-        </a>
-      </Link>
-      <nav>
-        <StTabList>
-          <StTab isActive={router.pathname === '/home'}>
-            <Link href="/home">
-              <a>홈</a>
-            </Link>
-          </StTab>
-          <StTab isActive={router.pathname === '/learn'}>
-            <Link href="/learn">
-              <a>학습하기</a>
-            </Link>
-          </StTab>
-          <StTab isActive={router.pathname === '/review'}>
-            <Link href="/review">
-              <a>복습하기</a>
-            </Link>
-          </StTab>
-        </StTabList>
-      </nav>
-      <StLogin>로그인</StLogin>
-    </StNavigationBar>
+    <>
+      <StNavigationBar>
+        <Link href="/home">
+          <a>
+            <ImageDiv className="logo" priority src={imgLogo} layout="fill" alt="DeliverBle" />
+          </a>
+        </Link>
+        <nav>
+          <StTabList>
+            <StTab isActive={router.pathname === '/home'}>
+              <Link href="/home">
+                <a>홈</a>
+              </Link>
+            </StTab>
+            <StTab isActive={router.pathname === '/learn'}>
+              <Link href="/learn">
+                <a>학습하기</a>
+              </Link>
+            </StTab>
+            <StTab isActive={router.pathname === '/review'}>
+              <Link href="/review">
+                <a>복습하기</a>
+              </Link>
+            </StTab>
+          </StTabList>
+        </nav>
+        <StLogin onClick={() => setIsModalOpen(true)}>로그인</StLogin>
+      </StNavigationBar>
+      {isModalOpen && <LoginModal closeModal={() => setIsModalOpen(false)} />}
```

Comment:
```suggestion
      {isModalOpen && <LoginModal setIsModalOpen ={setIsModalOpen}  />}
```
저는 set함수를 보통 요렇게 넘겨주는데 하위 컴포넌트에서도 set함수인게 보여서 좋더라구요 
다들 어떻게 하는 지 궁금하네용